packages/core/src/web/app/pages/ConnectMachineIp/hooks/useConnectionTest.ts

```diff
@@ -0,0 +1,280 @@
+              testState: countDown.current > 0 ? TestState.CONNECTION_TESTING : TestState.CONNECTION_TEST_FAILED,
+            }));
+
+            if (countDown.current <= 0) {
+              clearInterval(intervalId.current);
+              resolve(null);
+            }
+          }
+        } else {
+          clearInterval(intervalId.current);
+          resolve(null);
+        }
+      }, 1000) as unknown as number;
+    });
+  };
+
+  const testCamera = async (device: IDeviceInfo) => {
+    updateTestState({ testState: TestState.CAMERA_TESTING });
+
+    const res = await checkCamera(device);
+
+    if (res) {
+      updateTestState({ testState: TestState.TEST_COMPLETED });
+
+      return;
+    }
+
+    updateTestState({ testState: TestState.CAMERA_TEST_FAILED });
+    alertCaller.popUp({
+      caption: lang.initialize.connect_machine_ip.check_camera,
+      message: isPromark ? lang.web_cam.no_device : lang.topbar.alerts.fail_to_connect_with_camera,
+      messageIcon: 'warning',
+    });
+  };
+
+  const checkSwiftrayConnection = () => {
+    const { readyState } = swiftrayClient;
+
+    updateTestState({ testState: TestState.IP_TESTING });
+
+    if (readyState === WebSocket.OPEN) {
+      return true;
+    }
+
+    if (readyState === WebSocket.CLOSED) {
+      updateTestState({ testState: TestState.IP_UNREACHABLE });
+      alertCaller.popUp({
+        caption: lang.initialize.connect_machine_ip.check_swiftray_connection,
+        message: lang.initialize.connect_machine_ip.alert.swiftray_connection_error,
+        messageIcon: 'warning',
+      });
+    }
+
+    return false;
+  };
+
+  const handleStartTestForPromark = async () => {
+    if (!checkSwiftrayConnection()) {
+      return;
+    }
+
+    const device = await testConnection(({ model }) => promarkModels.has(model));
+
+    if (!device) {
+      alertCaller.popUp({
+        caption: lang.initialize.connect_machine_ip.check_connection,
+        message: lang.message.unknown_device,
+        messageIcon: 'warning',
+      });
+
+      return;
+    }
+
+    // for correctly select promark device serial
+    const res = await deviceMaster.select(device);
+
+    console.log('🚀 ~ useConnectionTest.ts:199 ~ handleStartTestForPromark ~ res:', res);
+
+    await testCamera(device);
+  };
+
+  const handleStartTest = async () => {
+    if (isPromark) {
+      handleStartTestForPromark();
+
+      return;
+    }
+
+    const { testState } = state;
+
+    if (isTesting(testState)) {
+      return;
+    }
+
+    if (!validateIpFormat()) {
+      return;
+    }
+
+    const ip = await testIpReachability();
+
+    if (!ip) {
+      return;
+    }
+
+    setUpLocalStorageIp(ip);
+
+    testingIps.forEach((testingIp) => {
+      discoverer.poke(testingIp);
+      discoverer.pokeTcp(testingIp);
+      discoverer.testTcp(testingIp);
+    });
+
+    const device = await testConnection();
+
+    if (!device) {
+      return;
+    }
+
+    if (!checkSoftwareForAdor(device)) {
```

Comment:
the condition inside `checkSoftwareForAdor` may need to be changed to `adorModels` instead of `modelsWithModules`